src/Data/Vec/Functional/Relation/Binary/Permutation/Properties.agda

```diff
@@ -36,10 +38,32 @@ private
 ↭-sym : Symmetric (Vector A) _↭_
 proj₁ (↭-sym (xs↭ys , _)) = flip xs↭ys
 proj₂ (↭-sym {x = xs} {ys} (xs↭ys , xs↭ys≡)) i = begin
-  ys (flip xs↭ys ⟨$⟩ʳ i)             ≡˘⟨ xs↭ys≡ _ ⟩
+  ys (flip xs↭ys ⟨$⟩ʳ i)              ≡⟨ xs↭ys≡ _ ⟨
   xs (xs↭ys ⟨$⟩ʳ (flip xs↭ys ⟨$⟩ʳ i)) ≡⟨ cong xs (inverseʳ xs↭ys) ⟩
-  xs i ∎
+  xs i                                ∎
+  where open ≡-Reasoning
 
 ↭-trans : Transitive (Vector A) _↭_
 proj₁ (↭-trans (xs↭ys , _) (ys↭zs , _))   = ys↭zs ∘ₚ xs↭ys
 proj₂ (↭-trans (_ , xs↭ys) (_ , ys↭zs)) _ = trans (xs↭ys _) (ys↭zs _)
+
+------------------------------------------------------------------------
+-- Structure
+
+isIndexedEquivalence : IsIndexedEquivalence (Vector A) _↭_
+isIndexedEquivalence {A = A} = record
+  { refl = ↭-refl
+  ; sym = ↭-sym
+  ; trans = λ {n₁ n₂ n₃} {xs : Vector A n₁} {ys : Vector A n₂} {zs : Vector A n₃}
```

Comment:
This looks very odd - defining `i` twice? I guess it is there because Agda can't infer all the implicits?

Reply:
Indeed, Agda is maddeningly obscure if you simply try to define
```agda
; trans = ↭-trans
```
because it can't easily infer the implicit functional arguments which are the `Vector A`s `xs`, `ys`, `zs`. And they, in turn, because `Relation.Binary.Indexed.Heterogeneous.Definitions` uses full generalised transitivity, need distinct length indices...

Then there's the `i` telescopic binding, seemingly appearing twice: the first is because the `Indexed` machinery expects an implicit index `i` at which to begin a term of type generalised transitivity (I think we could make a renaming change in `Relation.Binary.Indexed.Heterogeneous.Definitions` to tweak this binding), but then the second (distinct!) binding arises because while in `Relation.Binary.Definitions` (to which the above module delegates), `Reflexivity`, `Symmetry` use `x` and `y` for their implicit arguments, a legacy we have *never* attempted to change is that `Transitivity` uses `i`, `j`, `k` for *its* implicit arguments!!! Frankly, it's a total nightmare, and if I could face the knock-on viscosity of changing `Relation.Binary.Definitions.Trans(itivity)` to use `x`, `y`, `z` then I would...
... the only saving grace is that Agda accepts (and expects) the two telescopic bindings to be given separately, even though they ostensibly have the same name. But as a ux 'bug', it takes some beating as to its 'wickedness'. 

Honestly, if you can figure out a simpler version/solution, please show me!

Reply:
I can see why Agda can't do the full inference, I think. Yes, changing `Transitivity` to `x`, `y`, `z` might be a good idea.

But now I see that such a change shouldn't hold up this one.

Reply:
Thanks for your understanding!